src/NaCl.Core/Base/ChaChaIntrinsics/ChaCha512.cs

```diff
@@ -0,0 +1,244 @@
+﻿#if INTRINSICS
```

Comment:
I wonder if we do need the preprocessor directive since we are checking if the hardware supports it